rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,181 @@
+code / PRs (Pull Requests) to Apache TVM.
+
+# Motivation
+[motivation]: #motivation
+
+Currently TVM commit logs are less than ideal because many commit messages lack
+valuable information and don't follow any format standard.
+
+Valuable information is usually left behind in Github PR conversations or
+discussion threads in the Discuss forum, making it hard to retrieve them when
+inspecting the commit messages -- using `git log`, for instance.
+
+Because commit messages are an indirect but important aspect of code quality,
+and also important for code maintenance, it is essential for a long term open
+source project to ensure that they meet high standards.
+
+The importance of commit messages conveying enough context and information about
+the code being changed will grow as the project grows and bad (poorly written)
+commit messages can affect negatively the code quality of future changes that
+would otherwise benefit from past good commit messages if they existed.
+
+Beyond code itself, poorly written commit messages can also affect the community
+in other ways. For example, by not providing to new contributors a consistent
+and complete history or context for the code changes, it can work as a barrier
+for new contributions because much more time will be necessary trying to
+understand what motivated a past critical but unclear change.
+
+Hence this Commit Message Guideline can help contributors to write good commit
+messages and so improve the current situation regarding the TVM commit logs.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Commit Message Guideline
+
+Apache TVM uses the Github (GH) platform for patch submission and code review
+via Pull Requests (PRs). The final commit (title and body) that is merged into
+the Apache TVM main tree is composed of the PR's title and body and must be kept
+updated and reflecting the new changes in the code as per the reviews and
+discussions.
+
+Although these guidelines apply essentially to the PRs’ title and body messages,
+because GH auto-generates the PR’s title and body from the commits on a given
+branch, it’s recommended to follow these guidelines right from the beginning,
+when preparing commits in general to be submitted to the Apache TVM project.
+This will ease the creation of a new PR, avoiding rework, and also will help the
+review.
+
+The rules below will help to achieve uniformity that has several benefits, both
+for review and for the code base maintenance as a whole, helping you to write
+commit messages with a good quality suitable for the Apache TVM project,
+allowing fast log searches, bisecting, and so on.
+
+_PR/commit title_:
+
+* Guarantee a title exists (enforced);
+* Don’t use Github usernames in the title, like @username (enforced);
+* Check if a tag should be present as a hint about what component(s) of the code
+  the commits “touch”. For example [BugFix], [CI], [microTVM], and [TVMC]. Tags
```

Comment:
I'm open to suggestion. I don't have any strong idea other than trying to establish a common set of acceptable tags, -- if that's even possible, but not sure how.

Reply:
I would do snake case and multiple brackets `[BugFix][CI]` as those seem to be the most used (from skimming the first page of open PRS).

Reply:
@tkonolige I'll added that to the CMG (Commit Message Guideline). But just to clarify, did you mean camel case (or even upper camel case) instead of snake case? I think snake case would be for instance [bug_fix]...

Reply:
Whoops I meant upper camel case.

Reply:
Addressed here: https://github.com/apache/tvm-rfcs/pull/88/commits/8486cc76cd7b76b244af02b561c889bfce95ea37

Reply:
@leandron @tkonolige Tags are marked as "enforced" I'm not totally convinced of it, but I don't strongly oppose to that. If we want to keep this way I think just the presence of tags must be enforced, not the capitalization etc. Also, the bullet on tags starts with "Check if a tag should be present as a hint...", if it's indeed enforced that must be chanced, since it would not be optional. PTAL.

Reply:
@driazati can you comment?

Reply:
I don't think capitalization necessarily matters, it would just be nice to have a recommendation.

Reply:
@tkonolige Thanks for the clarification. @leandron since all the approvals for this RFC was with the text that had tag presence as "enforced" I think are all aligned that it should be kept as such -- like in the original proposal. But given @tkonolige comment above, I just made a bit clearer that the capitalization is not enforced with this commit here: https://github.com/apache/tvm-rfcs/pull/88/commits/32527cf74bbdb8999b5a6ccc417292c56f574dda So I'm marking this conversation as resolved. If that's not the case, please reopen. Thanks!